plugins/inputs/directory_monitor/directory_monitor.go

```diff
@@ -256,16 +256,14 @@ func (monitor *DirectoryMonitor) ingestFile(filePath string) error {
 
 func (monitor *DirectoryMonitor) parseFile(parser parsers.Parser, reader io.Reader) error {
 	// Read the file line-by-line and parse with the configured parse method.
-	firstLine := true
+	lineNumber := 0
 	scanner := bufio.NewScanner(reader)
 	for scanner.Scan() {
-		metrics, err := monitor.parseLine(parser, scanner.Bytes(), firstLine)
+		metrics, err := monitor.parseLine(parser, scanner.Bytes(), lineNumber)
```

Comment:
This is a pure workaround around the linter. You are doing the same as before just silencing the linter by exploiting its dumbness. :-) Shall we really do this or better switch off that rule?

Reply:
I know :) I don't have better idea, how to fix this in the better way without reorganizing a lot of code. 
Rule itself is not that dumb, for other cases (in this PR and previous ones) it was pretty easy to fix this kind of finding (see: https://github.com/influxdata/telegraf/pull/9379/files#diff-8f6f90da1087a523fbc9bb69df6736bf68545e6d9fef07b5537b9ce22ff4a257L886 )

Reply:
@srebhan 
`flag-parameter` rule disabled, `directory_monitor.go` reverted